File: 0463-island-perimeter/0463-island-perimeter.cpp
class Solution {
public:
    int islandPerimeter(vector<vector<int>>& grid) {
        int r = grid.size();
        int c = grid[0].size();
        int v = 0;
        
        for (int i = 0; i < r; i++) {
            for (int j = 0; j < c; j++) {
                if (grid[i][j] == 1) {
                    int up = (i == 0) ? 0 : grid[i-1][j];
                    int down = (i == r-1) ? 0 : grid[i+1][j];
                    int left = (j == 0) ? 0 : grid[i][j-1];
                    int right = (j == c-1) ? 0 : grid[i][j+1];
                    
                    v += 4 - (up + down + left + right);
                }
            }
        }
        
        return v;
    }
};

File: 0463-island-perimeter/0463-island-perimeter.java
class Solution {
    public int islandPerimeter(int[][] grid) {
        int rows = grid.length;
        int columns = grid[0].length;
        int u,d,l,ri;
        int t = 0;
        
        for(int r=0;r<rows;r++){
            
            for(int c=0;c<columns;c++){
                
                if(grid[r][c]==1){
                    if(r==0){
                        u = 0;
                    }
                    else{
                        u = grid[r-1][c];
                    }
                    
                    if(c==0){
                        l = 0;
                    }
                    else{
                        l = grid[r][c-1];
                    }
                    
                    if(r==rows-1){
                        d = 0;
                    }
                    else{
                        d = grid[r+1][c];
                    }
                    
                    if(c==columns-1){
                        ri= 0;
                    }
                    else{
                        ri = grid[r][c+1];
                    }
                    
                    t  = t + 4 - (u + d + l + ri);
                }
            }
        }
        return t;
    }
}